pkg/client/unversioned/helper.go

```diff
@@ -290,7 +291,8 @@ func SetKubernetesDefaults(config *Config) error {
 	if len(config.UserAgent) == 0 {
 		config.UserAgent = DefaultKubernetesUserAgent()
 	}
-	if len(config.Version) == 0 {
+	if len(config.Version) == 0 || apiutil.GetGroup(config.Version) != "" {
```

Comment:
If we don't do this, the code will always fail [here](https://github.com/kubernetes/kubernetes/pull/14461/files#diff-f013b1aafa99ed0c479d939400d636e9R299) unless config.Version is either not set or set to `v1`. Users still need a RESTClient set with `SetKubernetesDefaults` (see [here](https://github.com/kubernetes/kubernetes/pull/14461/files#diff-f013b1aafa99ed0c479d939400d636e9R140)) even when they're using api groups other than `""` (api/v1). 

cc @caesarxuchao